driver/src/orderbook/streamed/block_timestamp.rs

```diff
@@ -21,3 +21,36 @@ impl BlockTimestamp for Web3 {
         .boxed()
     }
 }
+
+/// A cache for the block timestamp which avoids having to query the node in the case where we
+/// receive multiple events from the same block in a row.
+#[derive(Debug)]
+pub struct BlockTimestampCache<T> {
+    block_timestamp: T,
+    hash: H256,
+    timestamp: u64,
+}
+
+impl<T> BlockTimestampCache<T> {
+    pub fn new(block_timestamp: T) -> Self {
+        Self {
+            block_timestamp,
+            hash: H256::zero(),
+            timestamp: 0,
```

Comment:
Is there on value in caching multiple blocks here instead of only the last one?

Reply:
No because events nearly always come in in order.

Reply:
When there is a reorg, we might get events again from a previous block, wouldn't we? And it's really not a lot of storage, is it? Are you worried about cache eviction logic?

Reply:
This optimization really only does something for the past events because there we have a long list of events in order and need the timestamp for all of them.
New events from the stream don't occur that often so it won't help much there. Reorgs occur rarely and still contain a small number of events compared to the past events.